daml-assistant/daml-helper/src/DamlHelper/Run.hs

```diff
@@ -84,53 +85,162 @@ data ReplaceExtension
     -- ^ Replace the extension if the current extension is newer.
     | ReplaceExtAlways
     -- ^ Always replace the extension.
+    | ReplaceExtPublished
+    -- ^ Replace with published extension (the default).
 
-runDamlStudio :: ReplaceExtension -> [String] -> IO ()
-runDamlStudio replaceExt remainingArguments = do
-    sdkPath <- getSdkPath
-    vscodeExtensionsDir <- fmap (</> ".vscode/extensions") getHomeDirectory
-    let vscodeExtensionName = "da-vscode-daml-extension"
-    let vscodeExtensionSrcDir = sdkPath </> "studio"
-    let vscodeExtensionTargetDir = vscodeExtensionsDir </> vscodeExtensionName
-    whenM (shouldReplaceExtension replaceExt vscodeExtensionTargetDir) $
-        removePathForcibly vscodeExtensionTargetDir
-    installExtension vscodeExtensionSrcDir vscodeExtensionTargetDir
-    -- Note that it is important that we use `shell` rather than `proc` here as
-    -- `proc` will look for `code.exe` in PATH which does not exist.
-    projectPathM <- getProjectPath
-    let codeCommand
-            | isMac = "open -a \"Visual Studio Code\""
-            | otherwise = "code"
-        path = fromMaybe "." projectPathM
-        command = unwords $ codeCommand : path : remainingArguments
-
+-- | Run VS Code command with arguments, returning the exit code, stdout & stderr.
+runVsCodeCommand :: [String] -> IO (ExitCode, String, String)
+runVsCodeCommand args = do
     -- Strip DAML environment variables before calling vscode.
     originalEnv <- getEnvironment
     let strippedEnv = filter ((`notElem` damlEnvVars) . fst) originalEnv
```

Comment:
Why do we need this?

Reply:
We've had issues before where `daml studio` sets `DAML_SDK_VERSION` (and other environment variables, but this is the main one), and then this gets stuck in vscode, so when the user subsequently opens another project without closing vscode in between, the extension calls `daml damlc ide` with the wrong environment (i.e. with `DAML_SDK_VERSION` already set). This prevents all of that, letting `daml damlc ide` determine all the relevant variables at a later time.

(The alternative, I guess, is stripping out any `DAML_*` from the env when calling `daml damlc ide` in vscode.)

Reply:
I prefer having it in Haskell than Javascript, so here looks like a good place - but i think your comment deserves to be in the code. I imagine we have a ticket number we could include in the code too?

Reply:
Perfectly happy to keep the logic here, I had simply forgotten why we needed it. So :+1: for a comment, I’m sure I’ll manage to forget it again :slightly_smiling_face:

Reply:
Added a comment and linked to issue #1666